polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -1314,6 +1368,17 @@ async fn handle_incoming_statement<Context>(
 			},
 		};
 
+	if per_session.is_disabled(&statement.unchecked_validator_index()) {
+		gum::debug!(
+			target: LOG_TARGET,
+			?relay_parent,
+			validator_index = ?statement.unchecked_validator_index(),
+			"Ignoring a statement from disabled validator."
+		);
+		modify_reputation(reputation, ctx.sender(), peer, COST_DISABLED_VALIDATOR).await;
```

Comment:
Again out of curiosity - I see `COST_DISABLED_VALIDATOR` is `CostMajor` which sounds as a serious punishment. If a validator is a little out of sync it can understand later that a another validator is disabled. Why do we want to decrease its reputation in this case?

Reply:
Out of sync should not be an issue with view updates - right?

Reply:
Given that we are extending disabled validators on active leaf updates ... I think @tdimitrov might be right that this could be slightly problematic. If we punish an honest node that is just slightly out of sync, then we could end up affecting consensus after all. Most importantly approval voting.

Reply:
Are you suggesting a `CostMinor` instead of `CostMajor` reputation change?